pkg/controller/statusmanager/status_manager.go

```diff
@@ -49,8 +53,43 @@ type StatusManager struct {
 	relatedObjects []configv1.ObjectReference
 }
 
-func New(client client.Client, name, version string) *StatusManager {
-	return &StatusManager{client: client, name: name, version: version}
+func New(client client.Client, mapper meta.RESTMapper, name, version string) *StatusManager {
+	return &StatusManager{client: client, mapper: mapper, name: name, version: version}
+}
+
+func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.ClusterOperator) {
+	var found bool = false
```

Comment:
minor nit: I'd rather see this variable defined inside the while loop, since it's not meant to carry value "between" executions of the outer-most loop. Defining it outside is confusing to me.

I understand that it's still correct. Just a bit confusing.